### PR TITLE
chore: document po create and get

### PR DESCRIPTION
**Pull request by GrantZukowski**
Adding what I could find from purchase order input and purchase order line item input.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## v0.3.16
+### Added in V1
+- Added `POST /v1/purchase_orders` to create a purchase order.
+- Added `GET /v1/purchase_orders/:id` to get a specific purchase order.
```
Comment (warmwaffles) [inline CHANGELOG.md:13]:
```suggestion
- Added `GET /v1/purchase_orders/:id` to get a specific purchase order.


```

File: vic.api.v1.yaml [hunk 2]
```diff
@@ -283,3 +339,95 @@ components:
...
+      required:
+      - status
+      properties:
+        external_id:
+          oneOf:
+            - ref: "#/components/schemas/ExternalId"
+          description: |
+            The external id set by the ERP system. It does not have to be a UUID
+            and can be any string desired.
+          maxLength: 255
```
Comment (warmwaffles) [inline vic.api.v1.yaml:353]:
```suggestion
          $ref: "#/components/schemas/ExternalId"
```

File: vic.api.v1.yaml [hunk 3]
```diff
@@ -283,3 +339,95 @@ components:
...
+          format: date-time
+          description: The date the purchase order was created.
+        status:
+          $ref: "#/components/schemas/Status"
+        items:
+          type: array
+          items:
+            $ref: "#/components/schemas/PurchaseOrderLineItem"
+            description: |
+              The line items of the purchase order.
```
Comment (warmwaffles) [inline vic.api.v1.yaml:391]: This has to live outside of the `items:` directive.

File: vic.api.v1.yaml [hunk 4]
```diff
@@ -283,3 +339,95 @@ components:
...
+          description: The line item total of the line item.
+        dimensions:
+          type: array
+          nullable: true
+          items: "#/components/schemas/ReferenceInput"
+    ReferenceInput:
+      type: object
+      description: |
+          A simple input object that just has an internal id and external id.
+          At least one must be present.
```
Comment (warmwaffles) [inline vic.api.v1.yaml:425]:
```suggestion
          A simple input object that just has an `internal_id` and 
          `external_id`. At least one must be present. If both are provided, 
          `internal_id` takes priority.
```

File: vic.api.v1.yaml [hunk 5]
```diff
@@ -283,3 +339,95 @@ components:
...
+      properties:
+        internal_id:
+          type: string
+          description: The internal id of the object.
+        external_id:
+          type: string
+          description: The external id of the object.
```
Comment (warmwaffles) [inline vic.api.v1.yaml:432]:
```suggestion
        external_id:
          $ref: "#/components/schemas/ExternalId"
```

File: vic.api.v1.yaml [hunk 6]
```diff
@@ -205,6 +254,13 @@ components:
                   $ref: "#/components/schemas/Error"
 
   schemas:
+    ExternalId:
+      description: |
+        The external id set by the ERP system. It does not have to be a UUID and
+        can be any string desired.
+      type: string
+      example: 21b31bc7-1267-4335-893c-d7fe4706a238
+      maxLength: 255
```
Comment (warmwaffles) [inline vic.api.v1.yaml:263]: What do you think about adding a definition about `InternalId`?

File: vic.api.v1.yaml [hunk 7]
```diff
@@ -25,6 +25,55 @@ tags:
...
+  "/purchase_orders/{id}":
+    get:
+      description: Use this request to get data for a single purchase order
+      summary: Get a purchase order
+      operationId: getPurchaseOrder
+      tags:
+        - purchase orders
+      parameters:
+        - $ref: '#/components/parameters/PathId'
+        - $ref: '#/components/parameters/UseSystem'
```
Comment (warmwaffles) [inline vic.api.v1.yaml:63]: I'm not a huge fan of doing parameters this way. It's not clear at a glance when you look at the spec, what the parameters are supposed to be.

```yaml
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: string
      - name: use_system
        in: query
        schema:
          $ref: "#/components/schemas/SystemType"
```

Then of course we need to add `SystemType` to the schemas

```yaml
SystemType:
  enum:
    - internal
    - external
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## v0.3.16
+### Added in V1
```
Comment (warmwaffles) [inline CHANGELOG.md:11]:
```suggestion
### V1
```

Comment (warmwaffles) [reply CHANGELOG.md:11]: I am going to update the change log to just follow how ecto does it's changelogs and don't really thing the subdivision of `add` `change` and `removed` sections. Just slam them all in a list with a prefix of `- Added blah` `- Removed blah` `- Changed blah`

File: vic.api.v1.yaml (hunk 3 above)
Comment (GrantZukowski) [reply vic.api.v1.yaml:390]: Updated, thank you

File: CHANGELOG.md (hunk 8 above)
Comment (GrantZukowski) [reply CHANGELOG.md:11]: Sounds good.

**Comment by GrantZukowski** [comment]
@warmwaffles I used all the types you had added recently.  They were basically the same as what I had written.

File: vic.api.v1.yaml [hunk 9]
```diff
@@ -234,6 +282,23 @@ components:
...
+    ExternalId:
+      description: |
+        The external id set by the ERP system. It does not have to be a UUID and
+        can be any string desired.
+      type: string
+      example: 21b31bc7-1267-4335-893c-d7fe4706a238
+      maxLength: 255
+    InternalId:
+      description: |
+        The internal id set by the Vic API.  Ids may be integers, UUIDs, or sha1.
```
Comment (warmwaffles) [inline vic.api.v1.yaml:294]:
```suggestion
        The internal id set by the Vic API. The ids provided may be integers, UUIDs, 
        or a different non numerical value.
```

**Review by warmwaffles — APPROVED** [review]

**Pull request merged**